ess/src/API/EMBC.ESS/Resources/Suppliers/SupplierRepository.cs

```diff
@@ -235,7 +235,7 @@ private static void RemoveTeamSuppliers(EssContext essContext, era_supplier exis
 
             foreach (var ts1 in existingSupplier.era_era_supplier_era_essteamsupplier_SupplierId)
             {
-                var currentTeamSupplier = updatedSupplier.era_era_supplier_era_essteamsupplier_SupplierId.SingleOrDefault(ts2 => ts2._era_essteamid_value == ts1._era_essteamid_value);
+                var currentTeamSupplier = updatedSupplier.era_era_supplier_era_essteamsupplier_SupplierId.FirstOrDefault(ts2 => ts2._era_essteamid_value == ts1._era_essteamid_value);
```

Comment:
singleOrDefault should still work, if not, it's a problem as there are 2 team supplier records linked to the same team

Reply:
I see, my test that was conflicting with SingleOrDefault was incorrect.